src/it/java/uk/gov/companieshouse/company_appointments/tests/CompanyAppointmentFullRecordRepositoryITest.java

```diff
@@ -0,0 +1,70 @@
+import org.bson.Document;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.SimpleMongoClientDatabaseFactory;
+import org.springframework.test.annotation.DirtiesContext;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import uk.gov.companieshouse.company_appointments.CompanyAppointmentsApplication;
+import uk.gov.companieshouse.company_appointments.model.data.DeltaAppointmentApiEntity;
+import uk.gov.companieshouse.company_appointments.repository.CompanyAppointmentFullRecordRepository;
+
+@Testcontainers
+@AutoConfigureMockMvc
+@SpringBootTest(classes = CompanyAppointmentsApplication.class, webEnvironment = SpringBootTest.WebEnvironment.MOCK)
+@DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
+class CompanyAppointmentFullRecordRepositoryITest {
+
+    private static final String APPOINTMENT_ID = "7IjxamNGLlqtIingmTZJJ42Hw9Q";
+
+    @Autowired
+    private CompanyAppointmentFullRecordRepository repository;
+
+    @Container
+    private static final MongoDBContainer mongoDBContainer = new MongoDBContainer("mongo:4.4");
+
+    @BeforeAll
+    static void start() throws IOException {
+        System.setProperty("spring.data.mongodb.uri", mongoDBContainer.getReplicaSetUrl());
+        MongoTemplate mongoTemplate = new MongoTemplate(new SimpleMongoClientDatabaseFactory(mongoDBContainer.getReplicaSetUrl()));
+        mongoTemplate.createCollection("delta_appointments");
+        mongoTemplate.insert(Document.parse(IOUtils.resourceToString("/delta-appointment-data.json", StandardCharsets.UTF_8)), "delta_appointments");
+        System.setProperty("company-metrics-api.endpoint", "localhost");
+    }
+
+    @DisplayName("Repository successfully updates the company name and status of an appointment")
+    @Test
+    void patchAppointmentNameStatus() {
+        // given
+        Instant at = Instant.now().truncatedTo(ChronoUnit.SECONDS);
+
+        // when
+        long result = repository.patchAppointmentNameStatus(APPOINTMENT_ID, "test name", "test status", at, "etag");
+
+        // then
+        assertEquals(1L, result);
+        Optional<DeltaAppointmentApiEntity> actual = repository.findById(APPOINTMENT_ID);
+        assertTrue(actual.isPresent());
+        assertEquals("test name", actual.get().getCompanyName());
+        assertEquals("test status", actual.get().getCompanyStatus());
+        assertEquals(at, actual.get().getUpdated().getAt());
+        assertEquals("etag", actual.get().getEtag());
+    }
+}
```

Comment:
Could a negative repository test case be added please?

Reply:
Addressed in latest commit